client/jetpack-connect/authorize.js

```diff
@@ -616,10 +617,11 @@ export class JetpackAuthorize extends Component {
 			return `/start/pressable-nux?blogid=${ clientId }`;
 		}
 
-		// If the redirect is part of the Jetpack Search purchase flow
-		const isSearch = this.props.selectedPlanSlug === 'jetpack_search';
-
-		if ( isSearch ) {
+		// If the redirect is part of a Jetpack plan or product go to the checkout page
+		const jetpackCheckoutSlugs = OFFER_RESET_FLOW_TYPES.filter( ( productSlug ) =>
+			productSlug.includes( 'jetpack' )
```

Comment:
`wpcom_search` is included in `OFFER_RESET_FLOW_TYPES`. We don't want to included that one in here.